app/lib/actions/katello/repository/check_matching_content.rb

```diff
@@ -23,8 +23,10 @@ def run
             yum_metadata_files = yum_metadata_files_match?(source_repo, target_repo)
             checksum_match = (target_repo.saved_checksum_type == source_repo.saved_checksum_type)
 
+            published = target_repo.backend_service(::SmartProxy.pulp_primary).published?
```

Comment:
This line is causing ```NoMethodError: undefined method `pulp3_support?' for nil:NilClass``` in some of the tests

Reply:
@ianballou ah yes, I suspected that approach was not ideal. I will see about passing a relevant smart proxy parameter into this action. Thanks!